src/daemon/common_daemon_test.h

```diff
@@ -27,12 +29,15 @@ struct CommonDaemonTest : public ::testing::Test {
         test_service = service;
         service->CountConnectAttempts(&connect_count);
         service->CountListenAttempts(&listen_count);
-        service->CallOnConnect([this](net::EndPointPtr, String*) {
+        service_callback(service);
```

Comment:
Added service_callback to run code once network_server was created, but before any other code had a chance to run. Used to call for Listen to imitate remote servers.
Otherwise races could occur as code run from daemon could reach to remote before test had a chance to call test_service methods.